### PR TITLE
Add Template & fix docs issues

### PR DESCRIPTION
**Pull request by nvthongswansea**

### CODE REVIEW

**Review by bkircher — APPROVED** [review]
LGTM

File: gridscale/resource_gridscale_template_test.go [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+		*object = foundObject
+
+		return nil
+	}
+}
+
+func testAccCheckGridscaleTemplateDestroyCheck(s *terraform.State) error {
+	client := testAccProvider.Meta().(*gsclient.Client)
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "gridscale_ipv4" {
```
Comment (itakouna) [inline gridscale/resource_gridscale_template_test.go:78]: I think `gridscale_ipv4` should be `gridscale_template`  resource.

Comment (nvthongswansea) [reply gridscale/resource_gridscale_template_test.go:78]: I've already fixed it in another branch (will open a PR after this PR)

Comment (bkircher) [reply gridscale/resource_gridscale_template_test.go:78]: :+1:

**Pull request merged**